website/static/js/pages/profile-settings-addons-page.js

```diff
@@ -89,6 +91,18 @@ for (var i=0; i < addonEnabledSettings.length; i++) {
    }
 }
 
+/* Before closing the page, Check whether the newly checked addon are updated or not */
+$(window).on('beforeunload',function() {
+    //new checked items but not updated
+    var checked = uncheckedOnLoad.filter($('#selectAddonsForm input:checked'));
```

Comment:
@lyndsysimon  I just updated the input for the filter method and resolved the unexpected warning message when submitting. Thanks!